### PR TITLE
[SYCL] Update the test

### PR DESCRIPTION
**Pull request by KornevNikita**
test-e2e/DeviceLib/exp/exp-std-complex-double-edge-cases.hpp

Removed redundant info, fixed the device code input data, turned off one case for win

### CODE REVIEW

File: sycl/test-e2e/DeviceLib/exp/exp-std-complex-edge-cases.hpp [hunk 1]
```diff
@@ -1,17 +1,10 @@
 // This test checks edge cases handling for std::exp(std::complex<T>) used
 // in SYCL kernels.
-//
-// REQUIRES: aspect-fp64
```
Comment (AlexeySachkov) [inline sycl/test-e2e/DeviceLib/exp/exp-std-complex-edge-cases.hpp:4]: Note for other reviewers who may also think that this is an incorrect change: this is an `.hpp` file that is used from `.cpp` file which is the actual test, so it is fine.

**Review by AlexeySachkov — APPROVED** [review]

**Review by steffenlarsen — APPROVED** [review]

File: sycl/test-e2e/DeviceLib/exp/exp-std-complex-edge-cases.hpp [hunk 2]
```diff
@@ -294,8 +289,12 @@ template <typename T> bool test() {
       } else if (std::isfinite(testcases[i].imag()) &&
                  std::abs(testcases[i].imag()) <= 1) {
         CHECK(!std::signbit(r.real()), passed, i);
+// TODO: This case fails on win. Need to investigate and remove this macro
+// check.
```
Comment (KornevNikita) [inline sycl/test-e2e/DeviceLib/exp/exp-std-complex-edge-cases.hpp:293]:
```suggestion
// check.
// CMPLRLLVM-62905
```

**Review by againull — APPROVED** [review]
